MetaCell/geppetto-netpyne
# change plot.include component

### This requires:
- sync with netpyne/metadata   (a plot include branch was created in netpyne to use travis-ci)
- https://github.com/MetaCell/NetPyNE-UI/pull/70

### About simConfig.analysis.plot.include
all plots have a field **include** (of type list) to select which cells to include in the plot. The field accepts the following data types:
- string.       -->     'all'  |  'allCells' | "my_population"  |  "another_population"
- number     -->.     1,2, 445.     (cells global id's)
- tuple         -->.     ( "popLabel",  [1, 2 45, 234] )

### Current behavior
Right now we have decided to use a ListComponent (accepting only global id's).


### Propose behavior 
This PR uses menuItems to guide the user through the different options:
![animation](https://user-images.githubusercontent.com/32278395/44555690-cfa5dd00-a70c-11e8-8823-4879fd3768cf.gif)
 
### Test
- Add some populations and assign them a size through **numCells** field (other options are not allowed because sizes are computed after instantiation)
- add a new plot and fill it up by clicking in the different menuItems
- type `simConfig.analyses["plotRaster"]["include"]` in the notebook to check the field.
- Other option would be to load tut3.py and then instantiate and run to check the plots contain only those cells defined in the **include** field.

---
### This branch was renamed from plotInclude to plotinclude to make use of travis-ci.
here are the comment from that previews branch:

@adrianq  comments:
@rodriguez-facundo I have dropped a few comments/questions. It seems to be working fine and it is a very nice and useful component in terms of the user experience. Quite impressed with the work and a bit concerned with the concatenated 'ifs' ;) I will give it a thought about how to make it a bit more generic and not so ad-hoc
